pkg/resources/common/configmap.go

```diff
@@ -117,6 +117,24 @@ func (r *Reconciler) meshConfig() string {
 	return string(marshaledConfig)
 }
 
+func (r *Reconciler) getLocalityLBConfiguration() *istiov1beta1.LocalityLBConfiguration {
+	var localityLbConfiguration *istiov1beta1.LocalityLBConfiguration
+
+	if !util.PointerToBool(r.Config.Spec.LocalityLB.Enabled) {
+		return localityLbConfiguration
+	}
+
+	if r.Config.Spec.LocalityLB != nil {
+		localityLbConfiguration = r.Config.Spec.LocalityLB.DeepCopy()
+		localityLbConfiguration.Enabled = nil
```

Comment:
Why does it need to be set to `nil`?

Reply:
Otherwise the `enabled: ` will be set to the meshconfig as well.